inc/classes/class-settings.php

```diff
@@ -0,0 +1,229 @@
+		}
+		add_options_page( 'WP Google Login', 'WP Google Login', 'manage_options', 'wp-google-login', [
+			$this,
+			'options_page'
+		] );
+	}
+
+	/**
+	 * Adds a "Settings" link to this plugin's entry on the plugin list.
+	 *
+	 * @param array  $links Array of links for plugin actions.
+	 * @param string $file Path to the plugin file relative to the plugins directory.
+	 *
+	 * @return array $links Array of links for plugin actions.
+	 */
+	public function plugin_action_links( $links, $file ) {
+		if ( 'wp-google-login/wp-google-login.php' === $file ) {
+			$links[] = "<a href='options-general.php?page=wp-google-login'>" . __( 'Settings', 'wp-google-login' ) . '</a>';
+		}
+
+		return $links;
+	}
+
+	/**
+	 * Add admin menu.
+	 *
+	 * @return void
+	 */
+	public function settings_init() {
+		register_setting( 'wp_google_login', 'wp_google_login_settings' );
+
+		add_settings_section(
+			'wp_google_login_section',
+			__( 'WP Google Login Settings', 'wp-google-login' ),
+			[ $this, 'settings_section_callback' ],
+			'wp_google_login'
+		);
+
+		add_settings_field(
+			'wp_google_login_client_id',
+			__( 'Client ID', 'wp-google-login' ),
+			[ $this, 'wp_google_login_client_id_render' ],
+			'wp_google_login',
+			'wp_google_login_section',
+			[ 'label_for' => 'client-id' ]
+		);
+
+		add_settings_field(
+			'wp_google_login_client_secret',
+			__( 'Client Secret', 'wp-google-login' ),
+			[ $this, 'wp_google_login_client_secret_render' ],
+			'wp_google_login',
+			'wp_google_login_section',
+			[ 'label_for' => 'client-secret' ]
+		);
+
+		add_settings_field(
+			'wp_google_login_whitelisted_domains',
+			__( 'Whitelisted Domains', 'wp-google-login' ),
+			[ $this, 'wp_google_login_whitelisted_domains_render' ],
+			'wp_google_login',
+			'wp_google_login_section',
+			[ 'label_for' => 'whitelisted-domains' ]
+		);
+
+		add_settings_field(
+			'wp_google_login_enable_registration',
+			__( 'Create user if none exists', 'wp-google-login' ),
+			[ $this, 'wp_google_login_enable_registrationr' ],
+			'wp_google_login',
+			'wp_google_login_section',
+			[ 'label_for' => 'enable-registration' ]
+		);
+	}
+
+	/**
+	 * Render Client ID settings field.
+	 *
+	 * @return void
+	 */
+	public function wp_google_login_client_id_render() {
+		$client_id = wp_google_login_get_client_id();
+		$disabled  = '';
+		if ( defined( 'WP_GOOGLE_LOGIN_CLIENT_ID' ) ) {
+			$disabled = 'disabled';
+		}
+		?>
+		<input type='text' name='wp_google_login_settings[client_id]' id="client-id" <?php echo esc_attr( $disabled ); ?> value='<?php echo esc_attr( $client_id ); ?>'>
+		<p class="description">
+		<?php
+		echo wp_kses_post( sprintf( 
+			'<p>%1s <a target="_blank" href="%2s">%3s</a>.</p>', 
+			esc_html__( 'Create oAuth Client ID and Client Secret at', 'wp-google-login' ), 
+			esc_url( 'https://console.developers.google.com/apis/dashboard' ),
+			esc_html__( 'console.developers.google.com', 'wp-google-login' ) 
```

Comment:
We don't need to allow translation of the domain.

Reply:
Revised